File: war2/server/trunk/Common/src/war2/common/XgameError.java
package war2.common;

/**
 * 游戏异常
 * 
 * @author haijiang
 * @since 2012/6/3
 *
 */
public class XgameError extends RuntimeException {
	/** serialVersionUID */
	private static final long serialVersionUID = -1001L;

	/**
	 * 类默认构造器
	 * 
	 */
	public XgameError() {
		super();
	}

	/**
	 * 类参数构造器
	 * 
	 * @param msg
	 * 
	 */
	public XgameError(String msg) {
		super(msg);
	}

	/**
	 * 类参数构造器
	 * 
	 * @param err
	 * 
	 */
	public XgameError(Throwable err) {
		super(err);
	}

	/**
	 * 类参数构造器
	 * 
	 * @param msg
	 * @param err
	 * 
	 */
	public XgameError(String msg, Throwable err) {
		super(msg, err);
	}
}

File: server/Common/src/xgame/common/XgameError.scala
package xgame.common

/**
 * 游戏异常
 *
 * @author haijiang
 * @since 2013/6/16
 *
 */
class XgameError(val msg : String = null, val err : Throwable = null) extends RuntimeException(msg, err) {
   /**
    * 类参数构造器
    *
    * @param err
    * @return
    *
    */
    def this(err : Throwable) = this(null, err)
}
